be/src/exec/vectorized/join_hash_map.tpp

```diff
@@ -393,12 +393,23 @@ Status JoinHashMap<PT, BuildFunc, ProbeFunc>::_probe_output(ChunkPtr* probe_chun
     bool to_nullable = _table_items->left_to_nullable;
 
     for (size_t i = 0; i < _table_items->probe_column_count; i++) {
-        SlotDescriptor* slot = _table_items->probe_slots[i];
-        auto& column = (*probe_chunk)->get_column_by_slot_id(slot->id());
-        if (!column->is_nullable()) {
-            _copy_probe_column(&column, chunk, slot, to_nullable);
+        HashTableSlotDescriptor hash_table_slot = _table_items->probe_slots[i];
+        SlotDescriptor* slot = hash_table_slot.slot;
+        auto &column = (*probe_chunk)->get_column_by_slot_id(slot->id());
+        if (hash_table_slot.need_output) {
+            if (!column->is_nullable()) {
+                _copy_probe_column(&column, chunk, slot, to_nullable);
+            } else {
+                _copy_probe_nullable_column(&column, chunk, slot);
+            }
         } else {
-            _copy_probe_nullable_column(&column, chunk, slot);
+            ColumnPtr default_column = ColumnHelper::create_column(slot->type(), column->is_nullable() || to_nullable);
+            if (_probe_state->match_flag == JoinMatchFlag::ALL_MATCH_ONE) {
```

Comment:
remove the if, direct use default_column->append_default(_probe_state->count);

Reply:
ok